### PR TITLE
feat: gene info cards

### PR DESCRIPTION
**Pull request by sienacizdziel**
#### Reviewers
**Functional:** 
frontend: @seve 
backend in #323 : @MDunitz 
**Readability:** 

---

## Depends on PRs: 
- [x] #319 (smoke test fix) 
- [x] #320 (design changes) 
- [x] #323 (backend)
- [x] #325 (edge case in cxg files)

## Changes
- create component for gene info pop-up
- fetch gene ensembl ID from annoMatrix for more accurate search 
- call gene info API endpoint in data portal backend, which requests info from NCBI's E-Utilities API
- edit gene component to include gene info button
- use SDS styling and typing for TS
- ensure smooth interactions between scatterplot and gene info pop-ups 
- e2e tests 

## Notes for Reviewers
For my intern project: gene info cards. This feature allows users to search gene information (e.g. gene description, summary, synonyms, etc) about a gene in question by clicking on an info circle next to a gene name. On info button click, a pop-up similar to the scatterplot feature appears. This works for genes in both quick gene and genesets.

### CODE REVIEW

File: client/src/components/geneExpression/index.tsx [hunk 1]
```diff
@@ -8,37 +8,77 @@ import QuickGene from "./quickGene";
...
+      const df: Dataframe = await annoMatrix.fetch("var", varIndex);
+      dfIds = await annoMatrix.fetch("var", "feature_id");
+      this.setState({
+        geneIds: dfIds.col("feature_id").asArray() as DataframeValue[],
+      });
+      this.setState({
+        geneNames: df.col(varIndex).asArray() as DataframeValue[],
+      });
+    } catch {
+      console.log("no feature ids!");
```
Comment (seve) [inline client/src/components/geneExpression/index.tsx:49]: Should this be a console log or an error toast?

**Review by seve — CHANGES_REQUESTED** [review]
WIP Review

File: client/src/components/geneExpression/gene.tsx [hunk 2]
```diff
@@ -83,29 +81,57 @@ class Gene extends React.Component<{}, State> {
...
+      gene,
+    });
+
+    const info = await actions.fetchGeneInfo(geneId);
+    if (!info) {
+      dispatch({
+        type: "open gene info",
+        gene,
+        url: "",
+        name: "failed",
```
Comment (seve) [inline client/src/components/geneExpression/gene.tsx:105]: Instead of overloading the name attribute, maybe we should add an error attribute that accepts null or a string

File: client/src/components/geneExpression/quickGene.tsx [hunk 3]
```diff
@@ -50,14 +51,27 @@ function QuickGene() {
...
         try {
           const df: Dataframe = await annoMatrix.fetch("var", varIndex);
-          setStatus("success");
+          let dfIds: Dataframe;
+          const geneIdCol = "feature_id";
+
+          // if feature id column is available in var
+          if (annoMatrix.getMatrixColumns("var").includes(geneIdCol)) {
+            dfIds = await annoMatrix.fetch("var", geneIdCol);
+            console.log("id, success");
```
Comment (seve) [inline client/src/components/geneExpression/quickGene.tsx]: left in debug code?

File: client/src/components/geneExpression/quickGene.tsx [hunk 4]
```diff
@@ -50,14 +51,27 @@ function QuickGene() {
...
+            setGeneIds(dfIds.col("feature_id").asArray() as DataframeValue[]);
+          } else {
+            console.error("Could not find feature IDs.");
+          }
+
+          setStatus("name, success");
```
Comment (seve) [inline client/src/components/geneExpression/quickGene.tsx:71]: why the change, just curious

File: client/src/components/leftSidebar/index.tsx [hunk 5]
```diff
@@ -45,6 +49,15 @@ class LeftSideBar extends React.Component {
...
           <DynamicScatterplot />
         ) : null}
+        {geneIsOpen ? (
+          <GeneInfo
+            geneSummary=""
+            geneName=""
+            gene=""
+            geneUrl=""
+            geneSynonyms={[]}
+          />
```
Comment (seve) [inline client/src/components/leftSidebar/index.tsx:59]: Weird, is TS requiring us to call this component with the props that will be populated with redux?

File: client/src/reducers/controls.ts [hunk 6]
```diff
@@ -1,3 +1,36 @@
+/* logic for minimizing and maximizing pop-ups */
+const minMaxPopUps = (
+  geneLevel: string,
+  geneIsMin: boolean,
```
Comment (seve) [inline client/src/reducers/controls.ts:4]: if min is short for minimized, let's expand it

**Review by seve — CHANGES_REQUESTED** [review]

File: client/src/components/geneExpression/quickGene.tsx (hunk 4 above)
Comment (sienacizdziel) [reply client/src/components/geneExpression/quickGene.tsx:71]: This code is only for datasets that already have the "feature_id" column. Otherwise, we'll call the API with only gene name, since we don't have gene IDs built into the dataset (more of the code is implemented in the other PR that just recently got merged to this one)

File: client/src/components/leftSidebar/index.tsx (hunk 5 above)
Comment (sienacizdziel) [reply client/src/components/leftSidebar/index.tsx:59]: Yeah I think it looks so bad :(

File: client/src/components/geneExpression/quickGene.tsx (hunk 3 above)
Comment (sienacizdziel) [reply client/src/components/geneExpression/quickGene.tsx]: I removed the console.log, but the rest I think is still important for datasets that contain feature_id columns

**Review by seve — APPROVED** [review]
